addons/mail/static/src/messaging/component/follow_button/follow_button.js

```diff
@@ -0,0 +1,93 @@
+odoo.define('mail.messaging.component.FollowButton', function (require) {
+'use strict';
+
+const useStore = require('mail.messaging.component_hook.useStore');
+
+const { Component } = owl;
+const { useState } = owl.hooks;
+
+class FollowButton extends Component {
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        this.state = useState({
+            /**
+             * Determine whether the unfollow button is hovered or not.
```

Comment:
The comment should be more about the "why": because when it is hovered we want to display a different text/style.

Reply:
You can ask the question "what would I break if I removed this state?", this is the answer. This also an answer to a previous question I had: "why not just do it in css?".